From 10b141aa89c42efab04e9ea77e4c52ed37621d28 Mon Sep 17 00:00:00 2001
From: Sil Vilerino <sivileri@microsoft.com>
Date: Wed, 3 Jan 2024 09:23:02 -0500
Subject: [PATCH 2/2] Revert "d3d12: Fix screen->winsys leak in d3d12_screen"

This reverts commit 81c8e89ff8e846839fffc1d103b2080bef5c1b5c.

Fixes: 81c8e89ff8e ("d3d12: Fix screen->winsys leak in d3d12_screen")
Cc: mesa-stable
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/26872>
(cherry picked from commit e11605321aaf94e537751b09f3ec5133c385cc83)
---
 .pick_status.json                          | 2 +-
 src/gallium/drivers/d3d12/d3d12_screen.cpp | 4 ----
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/gallium/drivers/d3d12/d3d12_screen.cpp b/src/gallium/drivers/d3d12/d3d12_screen.cpp
index ee74cf2b064..86faa25a074 100644
--- a/src/gallium/drivers/d3d12/d3d12_screen.cpp
+++ b/src/gallium/drivers/d3d12/d3d12_screen.cpp
@@ -735,10 +735,6 @@ d3d12_deinit_screen(struct d3d12_screen *screen)
       screen->dev->Release();
       screen->dev = nullptr;
    }
-   if (screen->winsys) {
-      screen->winsys->destroy(screen->winsys);
-      screen->winsys = nullptr;
-   }
 }
 
 void
-- 
2.40.1

